src/node_api.cc

```diff
@@ -120,13 +120,14 @@ struct napi_env__ {
 #define CHECK_NEW_FROM_UTF8_LEN(env, result, str, len)                   \
   do {                                                                   \
     auto str_maybe = v8::String::NewFromUtf8(                            \
-        (env)->isolate, (str), v8::NewStringType::kInternalized, (len)); \
+        (env)->isolate, (str), v8::NewStringType::kInternalized,         \
+        static_cast<int>(len));                                          \
```

Comment:
It would be nice to have a `static_assert(static_cast<int>(NAPI_AUTO_LENGTH) == -1)` somewhere

Reply:
@addaleax that is a good idea.

Do you think it would make sense to add it just after

namespace {
namespace v8impl {

in node_api.cc

along with a comment indicating the v8 implementation of n-api depends on it being true ?

Reply:
@mhdawson I like to put that near where it’s actually used (i.e. around here) but feel free to put it wherever you like :)

Reply:
I was thinking you had somewhere else in mind, I'm good with it going into the macro itself